File: portal/src/types/analytics.ts
export interface ScoreBreakdown {
  overall: number | null;
  sentiment: number | null;
  politeness: number | null;
  compliance: number | null;
  resolution: number | null;
  upselling: number | null;
  response_time: number | null;
  honesty: number | null;
}

export interface TrendPoint {
  period: string;
  overall: number | null;
  sentiment: number | null;
  politeness: number | null;
  compliance: number | null;
}

export interface TagStats {
  tag: string;
  count: number;
  label: string;
}

export interface AnalyticsSummary {
  total_slots: number;
  evaluated_slots: number;
  failed_slots: number;
  pending_slots: number;
  avg_scores: ScoreBreakdown;
  score_trend: TrendPoint[];
  tag_stats: TagStats[];
  total_swearing_incidents: number;
  total_off_topic_incidents: number;
  total_unavailable_items: number;
  unavailable_item_frequency: Record<string, number>;
  language_distribution: Record<string, number>;
  status_distribution: Record<string, number>;
  avg_duration_ms: number | null;
  avg_tokens_used: number | null;
  period_start: string | null;
  period_end: string | null;
}

export interface AnalyticsSummaryParams {
  date_from?: string;
  date_to?: string;
  terminal_id?: string;
}

export type { Aggregation, AggregationListParams, AggregationListResponse } from "./aggregation";

File: backend/app/schemas/analytics.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class TagStats(BaseModel):
    tag: str
    count: int
    label: str


class ScoreBreakdown(BaseModel):
    overall: float | None
    sentiment: float | None
    politeness: float | None
    compliance: float | None
    resolution: float | None
    upselling: float | None
    response_time: float | None
    honesty: float | None


class TrendPoint(BaseModel):
    period: str
    overall: float | None
    sentiment: float | None
    politeness: float | None
    compliance: float | None


class AnalyticsSummary(BaseModel):
    total_slots: int
    evaluated_slots: int
    failed_slots: int
    pending_slots: int

    avg_scores: ScoreBreakdown
    score_trend: list[TrendPoint]

    tag_stats: list[TagStats]
    total_swearing_incidents: int
    total_off_topic_incidents: int
    total_unavailable_items: int
    unavailable_item_frequency: dict[str, int]

    language_distribution: dict[str, int]
    status_distribution: dict[str, int]

    avg_duration_ms: float | None
    avg_tokens_used: float | None

    period_start: datetime | None
    period_end: datetime | None
